osu.Desktop/DiscordRichPresence.cs

```diff
@@ -99,6 +100,22 @@ private void updateStatus()
             {
                 presence.State = truncate(activity.Value.Status);
                 presence.Details = truncate(getDetails(activity.Value));
+
+                if (getBeatmap(activity.Value) is IBeatmapInfo beatmap && beatmap.OnlineID > 0)
+                {
+                    presence.Buttons = new[]
+                    {
+                        new Button
+                        {
+                            Label = "View beatmap",
+                            Url = $@"{api.WebsiteRootUrl}/beatmapsets/{beatmap.BeatmapSet?.OnlineID}#{ruleset.Value.ShortName}/{beatmap.OnlineID}"
```

Comment:
Interesting curveball here: what should happen if `ruleset.Value` is a custom ruleset and the current beatmap is converted to it? The constructed link will technically not be valid, but it appears that web has some sort of safety against this and will redirect to the correct URL somehow (using beatmap set/beatmap IDs, I presume)?

This *works* and doesn't seem to break, so I'm approving, but I'm not sure it should be allowed, so I'm going to keep away from the merge button for the time being.

Reply:
I'd hope web can support it, although in my testing it didn't seem to perform the difficulty selection correctly with a bad ruleset name.